airflow/api_fastapi/routes/public/dag_run.py

```diff
@@ -57,3 +57,19 @@ async def delete_dag_run(dag_id: str, dag_run_id: str, session: Annotated[Sessio
         )
 
     session.delete(dag_run)
+
+
+@dag_run_router.patch("/{dag_run_id}", responses=create_openapi_http_exception_doc([400, 401, 403, 404]))
```

Comment:
In the legacy api this is a `PUT` I believe. Also it is really dedicated to the `update_dag_run_state`. It is not a general 'updage_dag_run`.

`@dag_run_router.put`

A general `PATCH` would be missing the `update_mask` etc... (cf legacy and other PATCH endpoints)

Reply:
@pierrejeambrun , a bit confused about the HTTP verb here. From what I read([Here](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods))  a `PUT` is supposed to be for replacing entire content. a `PATCH` is supposed to be for making a partial modification. So, Shouldn't this be a PATCH, rather than a PUT since we are just modifying the state?

Also, According to [Link](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/PUT), a PUT should be Idempotent. Hence, partial replacements will cause the resource not to be identical(Especially if you consider a DAG object which has `last_parsed_time` , which keeps changing). 

Atleast from above, I feel like a PATCH should be having an update_mask, while a PUT should be used to completely replace the object. 



I apologise if I'm completely wrong. I've been reading about PUT vs PATCH as I did not know the difference

Reply:
The global idea yes:
- PUT a full object that override the existing or create a new one if it does not exist
- PATCH -> Update an object partially

By idempotent, what we mean is if you send the request multiple times it doesn't change the result. For instance sending mulitple times the request to set the state to `success` will just leave the task in success, so this would be idempotent.

Then there is how we defines things in airflow, indeed we patch partially a resource by setting just it's state field, so a PATCH would be more appropriate. And for `PATCH` we could just take a partial payload, and don't need a submask. But for now in airflow `PATCH` are defined with a submask and a full payload, and this set_state method used to be a `PUT` because `PATCH` requires both a full body and a `mask`.

I would say that at first we should just focus on migrating things over, so I would just do the same as the legacy code.

And in separate PR / discussions we can propose refactoring / improvement / breaking change that will go beyond the scope of this PR. (For instance if we decide that the `PATCH` can take partial body, then the `mask` is not needed anymore, and this route can become a standard `PATCH` which is more in line with the HTTP spec). But in a separate PR I would suggest, so we remain consistent for this one.

Reply:
I updated it as a PUT. But, in [legacy docs](https://airflow.apache.org/docs/apache-airflow/stable/stable-rest-api-ref.html#operation/update_dag_run_state), it still shows up as PATCH. should we update the docs?

Reply:
I think I just misread the spec. The legacy endpoint is indeed a 'patch' and we should keep it like that for now. We could make it consistent with others by adding an `update_mask` that only allow "state" do be upated. We have that for the patch Dag where only "is_paused" is editable.

Sorry for the confusion, I think your original idea of making that a "patch" was the best option.